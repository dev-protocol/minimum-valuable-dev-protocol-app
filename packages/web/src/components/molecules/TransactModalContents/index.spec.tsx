import React from 'react'
import { render } from '@testing-library/react'
import { TransactModalContents } from '.'
import Jdenticon from 'react-jdenticon'
import 'src/__mocks__/window/matchMedia.mock'

jest.mock('src/fixtures/dev-kit/hooks')
jest.mock('src/fixtures/wallet/hooks')
jest.mock('web3')
jest.mock('@dev/graphql')
jest.mock('react-jdenticon')

describe(`${TransactModalContents.name}`, () => {
  test('Snapshot', () => {
    ;(Jdenticon as jest.Mock).mockImplementation(() => <></>)
    const component = render(<TransactModalContents type="stake" propertyAddress="0x0" />)
    const tree = component.baseElement
    expect(tree).toMatchSnapshot()
  })
})
