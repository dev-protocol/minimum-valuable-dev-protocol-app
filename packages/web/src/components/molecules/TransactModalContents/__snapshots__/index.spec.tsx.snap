// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactModalContents Snapshot 1`] = `
<body>
  <div>
    <div
      class="qs0k37-0 jHBAYx"
    >
      <div
        class="qs0k37-1 fJfijo"
      >
        <div
          class="sc-2tazhb-0 xkRoZ"
        >
          <div
            class="sc-2tazhb-1 hZCnjb"
            size="60"
          />
        </div>
        <h4
          class="sc-1w8sy4l-3 qs0k37-3 cqQVgy jVYmtO"
        >
          0x0-name
        </h4>
        <span
          class="qs0k37-4 hwMEdI"
        >
          0x0
        </span>
      </div>
      <div
        class="sc-89297o-0 LrdOg"
      >
        <div
          class="sc-1d8aayd-1 lldfxv"
        >
          <span
            class="ant-input-group-wrapper ant-input-group-wrapper-lg ant-input-search ant-input-search-large ant-input-search-with-button sc-1d8aayd-0 dMeuEc"
          >
            <span
              class="ant-input-wrapper ant-input-group"
            >
              <span
                class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
              >
                <input
                  class="ant-input ant-input-lg"
                  id="stake"
                  type="number"
                  value=""
                />
                <span
                  class="ant-input-suffix"
                >
                  <button
                    class="sc-1ttu0x8-0 ibloSD"
                  >
                    DEV
                  </button>
                </span>
              </span>
              <span
                class="ant-input-group-addon"
              >
                <button
                  class="ant-btn ant-btn-primary ant-btn-lg ant-input-search-button"
                  type="button"
                >
                  <span>
                    Stake
                  </span>
                </button>
              </span>
            </span>
          </span>
        </div>
        <div
          style="height: 40px;"
        >
          <div
            class="sc-1mqmkj6-1 imJLIj"
          >
            <button
              aria-checked="false"
              class="ant-switch"
              role="switch"
              type="button"
            >
              <div
                class="ant-switch-handle"
              />
              <span
                class="ant-switch-inner"
              />
            </button>
            <div
              class="sc-1mqmkj6-0 izoQta"
            >
              <span
                aria-label="info-circle"
                class="anticon anticon-info-circle"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="info-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  />
                  <path
                    d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
            </div>
            <span
              style="font-size: 0.8em;"
            >
              Incognito Staking:
              <br />
              If this is ON, your staking will be hidden from the public.
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="ant-message"
    >
      <div>
        <div
          class="ant-message-notice"
        >
          <div
            class="ant-message-notice-content"
          >
            <div
              class="ant-message-custom-content ant-message-error"
            >
              <span
                aria-label="close-circle"
                class="anticon anticon-close-circle"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                  />
                </svg>
              </span>
              <span>
                fetch is not defined
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-message-notice"
        >
          <div
            class="ant-message-notice-content"
          >
            <div
              class="ant-message-custom-content ant-message-error"
            >
              <span
                aria-label="close-circle"
                class="anticon anticon-close-circle"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                  />
                </svg>
              </span>
              <span>
                fetch is not defined
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-message-notice"
        >
          <div
            class="ant-message-notice-content"
          >
            <div
              class="ant-message-custom-content ant-message-error"
            >
              <span
                aria-label="close-circle"
                class="anticon anticon-close-circle"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                  />
                </svg>
              </span>
              <span>
                fetch is not defined
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-message-notice"
        >
          <div
            class="ant-message-notice-content"
          >
            <div
              class="ant-message-custom-content ant-message-error"
            >
              <span
                aria-label="close-circle"
                class="anticon anticon-close-circle"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="close-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                  />
                </svg>
              </span>
              <span>
                fetch is not defined
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="ant-message"
    >
      <div />
    </div>
  </div>
  <div>
    <div
      class="ant-message"
    >
      <div />
    </div>
  </div>
  <div>
    <div
      class="ant-message"
    >
      <div />
    </div>
  </div>
</body>
`;
